networking.md

```diff
@@ -0,0 +1,43 @@
+# Networking
+
+Each machine (co-signing server, watchtower, wallet client) requires the capacity to connect to the others over TCP (mediated by the sync server operating a SOCKS5 proxy), and to engage in a noise_kk (or noise_xk) handshake (using pre-shared static public keys) to establish an authenticated and encrypted channel. 
+
+## API
+
+The first step is to develop a library that any component can use to participate securely in network communications. For this, the following API is proposed:
+
+`listen(ip_address, port)` - Start listening for TCP connections on ip_address:port
+
+`connect(ip_address, port)`   - Connect to ip_address:port
+
+These sockets/streams should be managed asyncronously to allow for handling multiple connections. Details of connection attempts should be logged.
+
+`ceremony_input(public_key, identifier)`  - Ceremony data input interface (static public keys for use in handshake)
+
+`handshake_initiate(ephemeral keypair, remote_static_public_key)`   - initiate handshake
+
+`handshake_respond(ephemeral keypair, remote_static_public_key)`   - respond to handshake initiation
+                
+The handshake ensures authenticated and encrypted communication (assuming trust in ceremony).
+
+> Secure key management (static private key) requires interface with dongle/HM
+
+> Deeper investigation into formal verification/ generation of noise_kk libs to ascertain suitability
+
+> Use of correct cryptographic primitives (SHA256 and secp256k1) [Change prims.rs and scan remaining lib code for conflicts]
```

Comment:
Do you mean modifying the crypto library ? If so, why ? Repeating myself re the discussion on MM, boutique-tweaking the crypto ourselves seems... to be **quite** a bad idea.

Reply:
The noise_kk lib generated at https://noiseexplorer.com/patterns/KK/ is generated from a formal spec and is verified. I think it's worthwhile considering it as opposed to re-purposing the rust-lightning implementation. As for required changes, I'm **not** advocating implementing cryptographic functions. Merely to point the primitives library to the appropriate functions (sha256 and secp256k1). This may be necessary for supporting available hardware modules.